### PR TITLE
ohlc: use all coords on given x-axis to compute min coord diff

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/1065

- ... instead of using only per-trace coords
- adding a loop over all traces in calcTransform is unfortunate,
  but is required as transform do not have access to the
  setPostiions step.

cc @alexcjohnson

### CODE REVIEW

**Comment by alexcjohnson** [comment]
Per our slack chat on Friday (edited bcs you convinced me it's per x-axis, not per subplot):

> what would happen if you for example display tickers for two stocks traded on opposite sides of the world, and to indicate that you offset one of them by 10 hours (or whatever) from the other.

> you wouldn’t want the minimum difference to drop to 10 hours, it should stay at 24 hours

> I feel like within each x axis you should calculate the minimum difference for each trace (ignoring traces with only one x value), then take the smallest of all of those as the winner for all traces on that x axis.

So, to also avoid looping over *all* traces during calc for *each* trace, I think we need to do something like:
- If this is calc for the *first* OHLC trace on this x axis, do the calculation and stash the result as a private attr in fullData for *all* the traces.
- If it's not the first one, just return the previously stashed result
- The calculation for a given trace should be the same as it was before (though I guess `Lib.distinctVals` is better because it sorts them... but we shouldn't concatenate into a single array)
- Then take the smallest nonzero difference for any trace, and that becomes the value for all of them.
- Can you make a test that works like my example, with a few hours offset between the traces? Jasmine should be fine, wouldn't need to be an image test.

File: test/jasmine/tests/finance_test.js [hunk 1]
```diff
@@ -564,6 +564,78 @@ describe('finance charts calc transforms:', function() {
...
+        });
+
+        // shift time coordinates by 10 hours
+        trace1.x = trace1.x.map(function(d) {
+            return d + ' 10:00';
+        });
+
+        var out = _calc([trace0, trace1]);
+
+        expect(out[0].x.map(ms2DateTime)).toEqual([
```
Comment (etpinard) [inline test/jasmine/tests/finance_test.js]: @alexcjohnson I think this is what you wanted. Can you confirm?

Comment (alexcjohnson) [inline test/jasmine/tests/finance_test.js]: Yes, looks great. Thanks!

**Comment by alexcjohnson** [comment]
💃

**Pull request merged**